src/attestation.rs

```diff
@@ -55,6 +59,74 @@ pub enum AttestationType {
     Uncertain(Credential),
 }
 
+// first Verification procedure if self attestation is in use.
+// Validate that alg matches the algorithm of the credentialPublicKey in authenticatorData.
+fn is_matching_agorithm(
+    att_stmt_map: &BTreeMap<ObjectKey, Value>,
+    alg_from_auhenticator_data: &COSEContentType,
+) -> bool {
+    att_stmt_map
+        .get(&serde_cbor::ObjectKey::String("alg".to_string()))
+        .and_then(|alg| alg.as_i64())
+        .map(|alg| alg == i64::from(alg_from_auhenticator_data))
+        .unwrap_or(false)
+}
+
+// https://w3c.github.io/webauthn/#sctn-packed-attestation
```

Comment:
It's worth taking the text from this section and turning them into comments to inline with the code, so that we can assert all the steps have been taken as per the specification. See verify_fidou2f_attestation for an example of how I did this.

Reply:
Good point, I'll add the verification steps from the spec into inline comment.